source/StarField.cpp

```diff
@@ -90,57 +90,61 @@ void StarField::SetHaze(const Sprite *sprite)
 
 void StarField::Draw(const Point &pos, const Point &vel, double zoom) const
 {
-	glUseProgram(shader.Object());
-	glBindVertexArray(vao);
+	// Draw the starfield unless it is disabled in the preferences.
+	if(Preferences::Has("Draw starfield"))
```

Comment:
I think you should do
```
if(!Preferences::Has("Draw starfield"))
	return;
```
 as to not need to indent everything. See lower down the same thing done for the background haze preference.